src/Workspaces/Core/Portable/Workspace/AdhocWorkspace.cs

```diff
@@ -201,5 +201,33 @@ public override void CloseAdditionalDocument(DocumentId documentId)
                 this.OnAdditionalDocumentClosed(documentId, loader);
             }
         }
+
+        /// <summary>
+        /// Puts the specified analyzer config document into the open state.
+        /// </summary>
+        public override void OpenAnalyzerConfigDocument(DocumentId documentId, bool activate = true)
+        {
+            var doc = this.CurrentSolution.GetAnalyzerConfigDocument(documentId);
+            if (doc != null)
+            {
+                var text = doc.GetTextSynchronously(CancellationToken.None);
+                this.OnAnalyzerConfigDocumentOpened(documentId, text.Container, activate);
+            }
+        }
+
+        /// <summary>
+        /// Puts the specified analyzer config document into the closed state
+        /// </summary>
+        public override void CloseAnalyzerConfigDocument(DocumentId documentId)
+        {
+            var doc = this.CurrentSolution.GetAnalyzerConfigDocument(documentId);
+            if (doc != null)
+            {
+                var text = doc.GetTextSynchronously(CancellationToken.None);
+                var version = doc.GetTextVersionSynchronously(CancellationToken.None);
+                var loader = TextLoader.From(TextAndVersion.Create(text, version, doc.FilePath));
+                this.OnAnalyzerConfigDocumentClosed(documentId, loader);
```

Comment:
another +1 to @jasonmalinowski  on my feedback about encapsulting doc handling in some way so that each new doc type doesn't require this sort of copy/paste.  (i know you're aware of the feedback, but these prs just make me sad).